js-plugin/src/main/kotlin/org/jetbrains/research/deepbugs/javascript/ide/inspections/base/JSDeepBugsBinExprInspection.kt

```diff
@@ -3,15 +3,15 @@ package org.jetbrains.research.deepbugs.javascript.ide.inspections.base
 import com.intellij.codeInspection.LocalInspectionToolSession
 import com.intellij.codeInspection.ProblemsHolder
 import com.intellij.lang.javascript.psi.JSBinaryExpression
-import com.intellij.psi.NavigatablePsiElement
-import org.jetbrains.research.deepbugs.javascript.datatypes.JSBinOp
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.research.deepbugs.javascript.datatypes.collect
 
-abstract class JSDeepBugsBinExprInspection : JSDeepBugsBaseInspection() {
-    abstract inner class JSDeepBugsBinOpVisitor(holder: ProblemsHolder, session: LocalInspectionToolSession) : JSDeepBugsVisitor(holder) {
-        override fun collect(node: NavigatablePsiElement, src: String) = JSBinOp.collectFromJSNode(node as JSBinaryExpression)
+abstract class JSDeepBugsBinExprInspection(threshold: Float = 0.8f) : JSDeepBugsBaseInspection<JSBinaryExpression>(threshold) {
+    override fun createVisitor(holder: ProblemsHolder, session: LocalInspectionToolSession): PsiElementVisitor = JSDeepBugsBinOpVisitor(holder)
 
+    inner class JSDeepBugsBinOpVisitor(holder: ProblemsHolder) : JSDeepBugsVisitor(holder) {
         override fun visitJSBinaryExpression(node: JSBinaryExpression?) {
```

Comment:
Can this node really be null? May we somehow check in visitor?

Reply:
This method overrides the default JSElementVisitor.visitJSBinaryExpression implementation, where the parameter doesn't have a @NotNull annotation